src/jit/importer.cpp

```diff
@@ -6958,7 +6958,7 @@ bool  Compiler::impMethodInfo_hasRetBuffArg(CORINFO_METHOD_INFO * methInfo)
     //
   #ifdef FEATURE_HFA
     // On ARM HFAs are returned in registers.
-    if (!info.compIsVarArgs && IsHfa(methInfo->args.retTypeClass))
+    if (!info.compIsVarArgs && !opts.compUseSoftFP && IsHfa(methInfo->args.retTypeClass))
     {
```

Comment:
This method Compiler::IsHfa answers the question is this class handle an HFA type.

For a varargs method the calling convention says to ignore HFA and pass them as if they weren't HFA's
And you can have a mix of methods some which are varargs and some that are not varargs.

But I think that with opts.compUseSoftFP you can't have this kind of mix of methods.
Basically if opts.compUseSoftFP is set when compiling a method it has to be on when compiling every method, 
as you are modifying the calling convention and both the called method and the callsite both have to agree how to pass and receive arguments.

My belief is that once compUseSoftFP is set it should always be set and you no longer have any HFA types at all.